android/src/main/java/com/reactnativestripesdk/StripeSdkModule.kt

```diff
@@ -39,12 +52,16 @@ class StripeSdkModule(reactContext: ReactApplicationContext, cardFieldManager: S
   private var confirmPaymentSheetPaymentPromise: Promise? = null
   private var presentPaymentSheetPromise: Promise? = null
   private var initPaymentSheetPromise: Promise? = null
-
   private var confirmPaymentClientSecret: String? = null
 
   private val mActivityEventListener = object : BaseActivityEventListener() {
     override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
-      if (::stripe.isInitialized) {
+      if (!::stripe.isInitialized) {
+        return
+      }
+      paymentSheetFragment?.activity?.activityResultRegistry?.dispatchResult(requestCode, resultCode, data)
+
+      if (VALID_REQUEST_CODES.contains(requestCode)) {
```

Comment:
Why do you need to filter?

Reply:
to make sure that the relevant `startActivityForResult` came from the expected source (stripe), and not a separate library that also uses activity triggers

Reply:
What prevents you from forwarding the onActivityResult to more than one function?  This function doesn't return if it has handled the activity result, and it doesn't include an else statement in the case where it is not one of the expected values.

Reply:
@michelleb-stripe sorry but I don't get what u mean, why should we return something there?

Reply:
You don't need to return something.  

I am not understanding why you need to filter at this point.  It is my understanding that the functions onActivityResult do their own filtering based on the result code, because it is each of the listeners that know the value of the result code they started.

If you call onActivityResult and the result is not for that handler it should ignore it.   In this way you could call multiple onActivityResult, and only the one listening for the result code would do the work.